src/renderer/api/kube-object.ts

```diff
@@ -86,6 +86,16 @@ export class KubeStatus {
   }
 }
 
+export interface KubeObjectStatus {
+  conditions: {
+    lastTransitionTime: string;
+    message: string;
+    reason: string;
+    status: string;
+    type?: string;
+  }[];
+}
```

Comment:
This is the starting of a similar type to `KubeObjectMetadata`. In the future we should be able to use it everywhere because kube uses this type for object (even custom objects) it seems.

Reply:
Are we sure all kubernetes objects have these?

Reply:
Also I think this interface does not match actual implementation, conditions seems to be an array of objects.

Reply:
So I am not sure if all kube controllers do use this (or something like this...)

After reading:
 - https://github.com/kubernetes/kubernetes/issues/50798
 - https://github.com/kubernetes/kubernetes/issues/7856
 - https://github.com/kubernetes/community/pull/4521

and https://maelvls.dev/kubernetes-conditions/

It seems clear that not everything does use this type (or at least MUST use something like this). But that it is still the recommended way of displaying the history of data.